contributors/design-proposals/apiserver-count-fix.md

```diff
@@ -0,0 +1,114 @@
+the Endpoints eventually consistent, but the code relies on the Endpoints
+count becoming equal to or greater than masterCount. When the apiservers
+become greater than the masterCount the Endpoints tend to flap.
+
+If the number endpoints were scaled down from automation, then the Endpoints
+would never become consistent. 
+
+## Proposal
+
+### Create New Reconciler
+
+| Kubernetes Release  | Quality | Description |
+| ------------- | ------------- | ----------- |
+| 1.9           | alpha         | <ul><li>Add a new reconciler</li><li>Add a command-line switch --new-reconciler</li><li>Add a command-line switch --old-reconciler</li></ul>
+| 1.10          | beta          | <ul><li>Turn on the new reconciler by default |</li></ul>
+| 1.11          | stable        | <ul><li>Remove code for old reconciler</li><li>Remove --old-reconciler</li><li>Remove --new-reconciler</li><li>Remove --apiserver-count</li></ul>
+
+The MasterCountEndpointReconciler does not meet the current needs for durability of API Endpoint creation, deletion, or failure cases.
+
+Create a new `MasterEndpointReconciler` within master/controller.go.
+
+Add a `kube-apiserver-endpoints-config` ConfigMap in the `default` namespace. The duration found within the map would be configurable by admins without a recompile. The ConfigMap would include the following:
+
+```go
+ConfigMap{
+	"expire-duration": "1m", // golang duration (ns,us,ms,s,m,h)
+}
+```
+
+Add a standard `kube-apiserver-endpoints` ConfigMap in the `default` namespace. The ConfigMap would be formed such that: 
+
+*Key Format*: ip-[IP String Formatted]-[port]
+
+```go
+ConfigMap{
+	"ip-2001-4860-4860--8888-443": "serialized JSON ControllerEndpointData",
+	"ip-192-168-0-3-443":          "serialized JSON ControllerEndpointData",
+}
+
+type ControllerEndpointData struct {
+     metav1.TypeMeta   `json:",inline"`
+     metav1.ObjectMeta `json:",inline"`
+     api.EndpointPort
+     CreationTimestamp Time `json:"creationTimestamp,omitempty" protobuf:"bytes,8,opt,name=creationTimestamp"`
+     UpdateTimestamp   Time `json:"creationTimestamp,omitempty" protobuf:"bytes,8,opt,name=updateTimestamp"`
+}
+```
+
+The reconcile loop will expire endpoints that do not meet the duration. On
+each reconcile loop (the loop runs every 10 seconds currently):
+
+1. Retrieve `kube-apiserver-endpoints-config` ConfigMap (as configMap)
+1. Retrieve `kube-apiserver-endpoints` ConfigMap (as endpointMap)
+1. Update the `UpdateTimestamp` for the currently running API server
+1. Remove all endpoints where the UpdateTimestamp is greater than `expire-duration` from the configMap.
+1. Write endpointMap back to Kubernetes ConfigMap API
+
+configmap.yml:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: kube-apiserver-endpoints
+  namespace: default
+```
+
+### Refactor Old Reconciler
```

Comment:
nit: Can you make it more obvious that these are the two current viable ways to do this? (and add one for the CRD thingy??)
like: `Possible solution 1: foo`, `Possible solution 2: bar`